sled-agent/src/sled_agent.rs

```diff
@@ -356,6 +400,37 @@ impl SledAgent {
             .await
             .map_err(Error::from)
     }
+
+    /// Issue a snapshot request for a Crucible disk attached to an instance
+    pub async fn instance_issue_disk_snapshot_request(
+        &self,
+        instance_id: Uuid,
+        disk_id: Uuid,
+        snapshot_id: Uuid,
+    ) -> Result<(), Error> {
+        self.instances
+            .instance_issue_disk_snapshot_request(
+                instance_id,
+                disk_id,
+                snapshot_id,
+            )
+            .await
+            .map_err(Error::from)
+    }
+
+    /// Issue a snapshot request for a Crucible disk not attached to an
+    /// instance.
+    pub async fn issue_disk_snapshot_request(
+        &self,
+        _disk_id: Uuid,
+        _volume_construction_request: VolumeConstructionRequest,
+        _snapshot_id: Uuid,
+    ) -> Result<(), Error> {
+        // For a disk not attached to an instance, implementation requires
+        // constructing a volume and performing a snapshot through some other
+        // means. Currently unimplemented.
+        todo!();
+    }
```

Comment:
@jclulow this is where I imagine something like your new service plugging in - it has to construct the volume and therefore requires the volume construction request.

Reply:
Nit: can we propagate an error upwards, rather than crashing here? This is an endpoint exposed to clients; if for some horrible reason it doesn't make it into v1, we shouldn't be crashing when this happens